src/ap/java/org/spongepowered/plugin/processor/PluginElement.java

```diff
@@ -181,6 +185,16 @@ void apply(Messager messager) {
                 this.metadata.loadAfter(new PluginMetadata.Dependency(id, dependency.version()), !dependency.optional());
             }
         }
+
+        if (!hasSpongeDependency) {
+            String version = Platform.API_VERSION.orElseThrow(
+                    () -> new IllegalStateException("Can't find SpongeAPI version"));
+            if (version.contains("-")) version = version.substring(0, version.indexOf('-'));
```

Comment:
Missing braces, also for the love of `<insert deity here>` just use a regex instead of this mess of ifs and splits.